src/plugins/hyperbee/hyperbee.cpp

```diff
@@ -0,0 +1,1247 @@
+//    json_object_object_add(json_object, "UserName", json_object_new_string(USERIDSTR(drakvuf)));
+    /* Process SessionID/UID */
+    json_object_object_add(json_object, "UserID", json_object_new_int(info->attached_proc_data.userid));
+    /* Process name */
+    json_object_object_add(json_object, "ProcessName", json_object_new_string(info->attached_proc_data.name));
+    json_object_object_add(json_object, "EventUID", json_object_new_int64(info->event_uid));
+    json_object_object_add(json_object, "CR3", json_object_new_string_fmt("0x%" PRIx64, info->regs->cr3));
+    json_object_object_add(json_object, "PageVA", json_object_new_string_fmt("0x%" PRIx64, page_va));
+    json_object_object_add(json_object, "VADBase", json_object_new_string_fmt("0x%" PRIx64, base_va));
+    json_object_object_add(json_object, "VADEnd", json_object_new_string_fmt("0x%" PRIx64, end_va));
+    json_object_object_add(json_object, "VADName", json_object_new_string(dll_name_str));
+    json_object_object_add(json_object, "DumpSize", json_object_new_string_fmt("0x%" PRIx64, dump_size));
+    json_object_object_add(json_object, "DumpFile", json_object_new_string(dump_file_name));
+    json_object_object_add(json_object, "SHA256", json_object_new_string(chk_str));
+    json_object_object_add(json_object, "DumpID", json_object_new_int(dump_id));
+    json_object_object_add(json_object, "TrapPA", json_object_new_string_fmt("0x%" PRIx64, info->trap_pa));
+    json_object_object_add(json_object, "GFN", json_object_new_string_fmt("0x%" PRIx64, info->trap->memaccess.gfn));
+    fprintf(fp, "%s\n", json_object_get_string(json_object));
+    fclose(fp);
+
+    json_object_put(json_object);
+}
+
+/**
+ * Method to get the SHA256 hash of a memory area. Similar to the dump_memory_region-code in the memdump plugin.
+ */
+const gchar* get_sha256_memory(
+    vmi_instance_t vmi,
+    access_context_t* ctx,
+    size_t len_bytes)
+{
+    void** access_ptrs = nullptr;
+
+    const gchar* chk_str = nullptr;
+    GChecksum* checksum = nullptr;
+
+    addr_t aligned_addr;
+    addr_t intra_page_offset;
+    size_t aligned_len;
+    size_t len_remainder;
+    size_t num_pages;
+    size_t tmp_len_bytes = len_bytes;
+
+    aligned_addr = ctx->addr & ~(VMI_PS_4KB - 1);
+    intra_page_offset = ctx->addr & (VMI_PS_4KB - 1);
+
+    aligned_len = len_bytes & ~(VMI_PS_4KB - 1);
+    len_remainder = len_bytes & (VMI_PS_4KB - 1);
+
+    if (len_remainder)
+    {
+        aligned_len += VMI_PS_4KB;
+    }
+
+    ctx->addr = aligned_addr;
+    num_pages = aligned_len / VMI_PS_4KB;
+
+    access_ptrs = (void**) g_try_malloc0(num_pages * sizeof(void*));
+    if (!access_ptrs)
+    {
+        goto error;
+    }
+
+    if (VMI_SUCCESS != vmi_mmap_guest(vmi, ctx, num_pages, access_ptrs))
+    {
+        PRINT_DEBUG("[HYPERBEE] Failed mmap guest\n");
+        goto error;
+    }
+
+    checksum = g_checksum_new(G_CHECKSUM_SHA256);
```

Comment:
`g_checksum_free()` needs to be called, or this leads to a memleak according to https://developer.gnome.org/glib/stable/glib-Data-Checksums.html#g-checksum-new

Reply:
Following your recommendation leads to a problem with double free: 

From get_sha256_memory() I'm returning chk_str:
https://github.com/tklengyel/drakvuf/blob/25e224fd37eb7d78f5e33ad7535c116c4832714b/src/plugins/hyperbee/hyperbee.cpp#L314
https://github.com/tklengyel/drakvuf/blob/25e224fd37eb7d78f5e33ad7535c116c4832714b/src/plugins/hyperbee/hyperbee.cpp#L326

And at the end of the calling method execute_faulted_cb() I free chk_str:
https://github.com/tklengyel/drakvuf/blob/25e224fd37eb7d78f5e33ad7535c116c4832714b/src/plugins/hyperbee/hyperbee.cpp#L965

I tried adding g_checksum_free() to get_sha256_memory() additionally but that leads to a double free.

How would you deal with that?

Reply:
¯\_(ツ)_/¯ I don't use glib as I use C++. Just smth I read in documentation. Summon @tklengyel?

Reply:
I improved this to use g_checksum_free().
I was not sure if the prior code was "wrong", but now it should be correct in any way.

Reply:
@BonusPlay Can you confirm this? Then I'd resolve this conversation. The checks were successful at that point:
https://github.com/tklengyel/drakvuf/pull/1133/checks?sha=8f1732c0c6c1f683d1a256aecf3b9affff6bc61e

Reply:
Honestly I have no idea, @icedevml / @chivay ?

Reply:
Calling `g_checksum_free` is correct as @BonusPlay pointed out. The double free is caused because of `g_free`.
From the documentation of `g_checksum_get_string`:
"the hexadecimal representation of the checksum. The returned string is owned by the checksum and should not be modified or freed."

Reply:
Exactly, that was what I found out as well. I'm not freeing chk_str anymore.
I create a copy of the checksum string (belonging to checksum) with g_strdup, and then free checksum.
And after printing the copied checksum string I am freeing that as well.